packages/runtime/src/controls/visitors/client-message-port-serialization-visitor.ts

```diff
@@ -0,0 +1,66 @@
+import {
+  AnyFunction,
+  ControlDefinition,
+  SerializedRecord,
+  serializeFunction,
+  serializeObject,
+  SerializationPlugin,
+  ControlSerializationVisitor,
+  isFunction,
+} from '@makeswift/controls'
+
+import { RichTextV2Definition } from '../rich-text-v2'
+
+export class ClientMessagePortSerializationVisitor extends ControlSerializationVisitor {
+  private transferables: Transferable[] = []
+
+  constructor() {
+    const serializeFunctionPlugin: SerializationPlugin<AnyFunction> = {
+      match: isFunction,
+      serialize: (val: AnyFunction) => {
+        const r = serializeFunction(val)
+        this.transferables.push(r)
+        return r
+      },
+    }
+
+    const serializeDefinitionPlugin: SerializationPlugin<ControlDefinition> = {
+      match: (val: unknown) => val instanceof ControlDefinition,
+      serialize: (val: ControlDefinition) => {
+        return val.accept(this)
+      },
```

Comment:
Super nitpicky, I'd just make this: `serialize: (val: ControlDefinition) => val.accept(this)`